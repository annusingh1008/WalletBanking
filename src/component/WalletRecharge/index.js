import React, { useState } from 'react'
import { Container, Row, Col, Form, Button, Modal } from 'react-bootstrap'
import { useDispatch, useSelector } from 'react-redux';
import { recharge } from '../../actions/recharge.actions';
import Layout from '../Layout'
import './style.css';

const WalletRecharge = () => {

  const [amount, setAmount] = useState();
  const [confirmModal, setConfirmModal] = useState(false);

  const dispatch = useDispatch();

  const auth = useSelector(state => state.auth)

  const confirmRecharge = (e) => {
    e.preventDefault();
    setConfirmModal(true);
  }

  const walletRecharge = () => {

    const rechargeDetails = {
      accountNumber: auth.user.accountNumber,
      amount: amount,
      currentAmount: auth.user.amount
    }

    dispatch(recharge(rechargeDetails));
  }

  return (
    <Layout sidebar>
      <Container className='container'>
        <Row md={12}>
          <Col>
            <div className='heading'>Wallet Recharge</div>
            <Form className='form' onSubmit={walletRecharge} >
              <label>Amount</label><br />
              <input
                type="text"
                placeholder='1000'
                value={amount}
                onChange={(e) => setAmount(e.target.value)}
              ></input><br />
              <Button type='variant'>Recharge</Button>
            </Form>
          </Col>
        </Row>

        {/* {confirmModal && (
          <Modal
            show={confirmModal}
            onHide={() => setConfirmModal(false)}
            backdrop="static"
            keyboard={false}
          >
            <Modal.Header closeButton>
              <Modal.Title>Confirm</Modal.Title>
            </Modal.Header>
            <Modal.Body>
              Are you sure
            </Modal.Body>
            <Modal.Footer>
              <Button
                variant="secondary"
                onClick={() => setConfirmModal(false)}
              >
                Close
              </Button>
              <Button variant="primary" onClick={walletRecharge} >Confirm</Button>
            </Modal.Footer>
          </Modal>
        )} */}
      </Container>
    </Layout>
  )
}

export default WalletRecharge
