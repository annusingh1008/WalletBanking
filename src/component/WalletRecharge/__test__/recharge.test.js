import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import WalletRecharge from "../index";
import { Provider } from "react-redux";
import store from "../../../store/index";
import { BrowserRouter as Router } from "react-router-dom";

const mockDispatch = jest.fn();
jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: () => mockDispatch,
}));

const localStorageData = {
  email: "annu",
  password: "1234",
};

const localStorageMock = (function () {
  return {
    getItem: function (key) {
      return localStorageData[key];
    },
    removeItem: function () {
      return null;
    },
    setItem: jest.fn(),
  };
})();

describe("WalletRecharge", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  Object.defineProperty(window, "localStorage", { value: localStorageMock });

  it("should render amount input element and label", () => {
    render(
      <Provider store={store}>
        <Router>
          <WalletRecharge />
        </Router>
      </Provider>
    );

    const inputElement1 = screen.getByPlaceholderText(/1000/i);
    expect(inputElement1).toBeInTheDocument();
  });

  it("should render amount label element", () => {
    render(
      <Provider store={store}>
        <Router>
          <WalletRecharge />
        </Router>
      </Provider>
    );

    const labeElement1 = screen.getByText("Amount");
    expect(labeElement1).toBeInTheDocument();
  });

  it("should render recharge button", () => {
    render(
      <Provider store={store}>
        <Router>
          <WalletRecharge />
        </Router>
      </Provider>
    );

    const buttonElement = screen.getByText("Recharge");
    expect(buttonElement).toBeInTheDocument();
  });

  it("renders properly", () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <WalletRecharge />
        </Router>
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  describe("#getItem", () => {
    it("returns the user email", () => {
      const result = localStorage.getItem("email");
      expect(result).toEqual("annu");
    });
  });

  it("should be able to type recharge details", () => {
    render(
      <Provider store={store}>
        <Router>
          <WalletRecharge />
        </Router>
      </Provider>
    );

    const textAreaElement1 = screen.getByPlaceholderText(/1000/i);
    fireEvent.change(textAreaElement1, { target: { value: "1000" } });
    expect(textAreaElement1.value).toBe("1000");
  });

  it("Form can be submitted", () => {
    const mockSubmit = jest.fn();
    const { queryByTestId } = render(
      <Provider store={store}>
        <Router>
          <WalletRecharge />
        </Router>
      </Provider>
    );
    fireEvent.submit(queryByTestId("button"));
    fireEvent.submit(queryByTestId("form"));
    expect(mockDispatch).toHaveBeenCalled();
  });
});
