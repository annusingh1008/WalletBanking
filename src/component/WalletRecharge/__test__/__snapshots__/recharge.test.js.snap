// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletRecharge renders properly 1`] = `
<nav
  class="navbar navbar-expand-lg navbar-dark bg-dark fixed-top"
  style="z-index: 2;"
>
  <div
    class="container-fluid"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      Wallet Banking
    </a>
  </div>
  <button
    aria-controls="responsive-navbar-nav"
    aria-label="Toggle navigation"
    class="navbar-toggler collapsed"
    type="button"
  >
    <span
      class="navbar-toggler-icon"
    />
  </button>
  <div
    class="navbar-collapse collapse"
    id="responsive-navbar-nav"
  >
    <div
      class="navbar-nav"
    >
      <li
        class="nav-item"
      >
        <span
          class="nav-link"
          data-tesid="signout"
        >
          Signout
        </span>
      </li>
    </div>
  </div>
</nav>
`;
